### PR TITLE
Conversions

### PR DESCRIPTION
**Pull request by ycscaly**
Implements conversion across different `Uxxx` types. This allows what was previously e.g.

```rust
let u: U128 = transcript.challenge(b"u");

let u: U256 = U128::ZERO.concat(&u);
let u: U512 = U256::ZERO.concat(&u);
let u: U1024 = U512::ZERO.concat(&u);
let u: U2048 = U1024::ZERO.concat(&u);
let u: U4096 = U2048::ZERO.concat(&u);
```

To become:
```rust
let u: U128 = transcript.challenge(b"u");
let u: U4096 = u.to_u4096;
```

### CODE REVIEW

**Comment by ycscaly** [comment]
Resolves #220

File: Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ rust-version = "1.65"
 
 [dependencies]
 subtle = { version = "2.4", default-features = false }
+paste = "1.0.12"
```
Comment (tarcieri) [inline Cargo.toml:21]: I would prefer not to add any additional mandatory dependencies.

Also, this use case doesn't seem to warrant an optional dependency either.

**Comment by tarcieri** [comment]
It seems like this would be better implemented in terms of the `From` trait.

**Comment by ycscaly** [comment]
> It seems like this would be better implemented in terms of the `From` trait.

Agreed

> It seems like this would be better implemented in terms of the `From` trait.

Agreed, this also allowed me to drop the `paste` dependency.

Should be good now

**Comment by ycscaly** [comment]
Replaced by #230

**Pull request closed without merging**